packages/@aws-cdk/toolkit-lib/lib/api/io/private/io-helper.ts

```diff
@@ -59,7 +59,7 @@ export class IoHelper implements IIoHost, IActionAwareIoHost {
    * Create a new marker from a given registry entry
    */
   public span<S extends object, E extends SpanEnd>(definition: SpanDefinition<S, E>) {
-    return new SpanMaker(this, definition);
+    return new SpanMaker(this, definition, (ioHost) => IoHelper.fromActionAwareIoHost(ioHost));
```

Comment:
This is needed because otherwise there's a circular dependency between `io-helper.ts` and `span.ts`. The alternative is one big file, but that's also nasty.